src/rust/src/x509/certificate.rs

```diff
@@ -319,6 +319,98 @@ impl Certificate {
             },
         )
     }
+
+    fn verify_signed_by<'p>(
+        &self,
+        py: pyo3::Python<'p>,
+        issuer: pyo3::PyRef<'_, Certificate>,
+    ) -> pyo3::PyResult<&'p pyo3::PyAny> {
+        let public_key = issuer.public_key(py)?;
+        let key_type = identify_public_key_type(py, public_key)?;
+        let hash_alg = self.signature_hash_algorithm(py)?;
```

Comment:
We need to check that the signature algorithm actually corresponds to the key type, don't we? Best to do that by OID I think.